File: Platforms/Bioma/BiomaSurfacePartonSoilSWATHourlyPartonC/src/java/BiomaSurfacePartonSoilSWATHourlyPartonC/SurfacePartonSoilSWATHourlyPartonCAuxiliary.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SurfacePartonSoilSWATHourlyPartonCAuxiliary
{
    private Double SurfaceTemperatureMaximum;
    private Double SurfaceTemperatureMinimum;
    private Double SurfaceTemperatureMaximum;
    private Double SurfaceTemperatureMinimum;
    
    public SurfacePartonSoilSWATHourlyPartonCAuxiliary() { }
    
    public SurfacePartonSoilSWATHourlyPartonCAuxiliary(SurfacePartonSoilSWATHourlyPartonCAuxiliary toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.SurfaceTemperatureMaximum = toCopy.getSurfaceTemperatureMaximum();
            this.SurfaceTemperatureMinimum = toCopy.getSurfaceTemperatureMinimum();
            this.SurfaceTemperatureMaximum = toCopy.getSurfaceTemperatureMaximum();
            this.SurfaceTemperatureMinimum = toCopy.getSurfaceTemperatureMinimum();
        }
    }
    public Double getSurfaceTemperatureMaximum()
    { return SurfaceTemperatureMaximum; }

    public void setSurfaceTemperatureMaximum(Double _SurfaceTemperatureMaximum)
    { this.SurfaceTemperatureMaximum= _SurfaceTemperatureMaximum; } 
    
    public Double getSurfaceTemperatureMinimum()
    { return SurfaceTemperatureMinimum; }

    public void setSurfaceTemperatureMinimum(Double _SurfaceTemperatureMinimum)
    { this.SurfaceTemperatureMinimum= _SurfaceTemperatureMinimum; } 
    
    public Double getSurfaceTemperatureMaximum()
    { return SurfaceTemperatureMaximum; }

    public void setSurfaceTemperatureMaximum(Double _SurfaceTemperatureMaximum)
    { this.SurfaceTemperatureMaximum= _SurfaceTemperatureMaximum; } 
    
    public Double getSurfaceTemperatureMinimum()
    { return SurfaceTemperatureMinimum; }

    public void setSurfaceTemperatureMinimum(Double _SurfaceTemperatureMinimum)
    { this.SurfaceTemperatureMinimum= _SurfaceTemperatureMinimum; } 
    
}

File: Models/BiomaSurfacePartonSoilSWATHourlyPartonC/src/cs/BiomaSurfacePartonSoilSWATHourlyPartonC/SurfacePartonSoilSWATHourlyPartonCAuxiliary.cs
using System;
using System.Collections.Generic;

public class SurfacePartonSoilSWATHourlyPartonCAuxiliary 
{
    private double _AboveGroundBiomass;
    private double[] _Sand;
    private double[] _OrganicMatter;
    private double _SurfaceSoilTemperature;
    private double _SurfaceTemperatureMinimum;
    private double _SurfaceTemperatureMaximum;
    
        public SurfacePartonSoilSWATHourlyPartonCAuxiliary() { }
    
    
    public SurfacePartonSoilSWATHourlyPartonCAuxiliary(SurfacePartonSoilSWATHourlyPartonCAuxiliary toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    AboveGroundBiomass = toCopy.AboveGroundBiomass;
    Sand = new double[toCopy.Sand.Length];
            for (int i = 0; i < toCopy.Sand.Length; i++)
            { Sand[i] = toCopy.Sand[i]; }
    
    OrganicMatter = new double[toCopy.OrganicMatter.Length];
            for (int i = 0; i < toCopy.OrganicMatter.Length; i++)
            { OrganicMatter[i] = toCopy.OrganicMatter[i]; }
    
    SurfaceSoilTemperature = toCopy.SurfaceSoilTemperature;
    SurfaceTemperatureMinimum = toCopy.SurfaceTemperatureMinimum;
    SurfaceTemperatureMaximum = toCopy.SurfaceTemperatureMaximum;
    }
    }
    public double AboveGroundBiomass
        {
            get { return this._AboveGroundBiomass; }
            set { this._AboveGroundBiomass= value; } 
        }
    public double[] Sand
        {
            get { return this._Sand; }
            set { this._Sand= value; } 
        }
    public double[] OrganicMatter
        {
            get { return this._OrganicMatter; }
            set { this._OrganicMatter= value; } 
        }
    public double SurfaceSoilTemperature
        {
            get { return this._SurfaceSoilTemperature; }
            set { this._SurfaceSoilTemperature= value; } 
        }
    public double SurfaceTemperatureMinimum
        {
            get { return this._SurfaceTemperatureMinimum; }
            set { this._SurfaceTemperatureMinimum= value; } 
        }
    public double SurfaceTemperatureMaximum
        {
            get { return this._SurfaceTemperatureMaximum; }
            set { this._SurfaceTemperatureMaximum= value; } 
        }
}